client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -64,8 +64,11 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 	}, [ location ] );
 
 	useEffect( () => {
-		recordStepStart( flow.name, kebabCase( currentRoute ), { intent } );
-	}, [ flow.name, currentRoute, intent ] );
+		// We record the event only when the step is not empty. Additionally, we should not fire this event whenever the intent is changed
+		if ( currentRoute ) {
+			recordStepStart( flow.name, kebabCase( currentRoute ), { intent } );
+		}
+	}, [ flow.name, currentRoute ] );
```

Comment:
Can we add a clearer comment on why we are leaving out `intent` from the hook's dependency list?
My suggestion:

```
// We leave out intent from the dependency list, due to the ONBOARD_STORE being reset in the exit flow.
// This causes the intent to become empty, and thus this event being fired again.
// eslint-disable-next-line react-hooks/exhaustive-deps
```